### ISSUE TITLE
Binding OnIdiom inside IsVisible causes crash in release mode

### ISSUE THREAD

**Issue by ghost** [dotnet/maui#16725]
### Description

If we use binding to a IsVisible property along with OnIdiom markup extension app crashes in release mode at InitiliazeCompenent(). Works fine on debug mode.

Ex : 
`
<Label Text="im hidden" IsVisible="{OnIdiom Default=false,Phone={Binding EnableLabel}}"/>
`

### Steps to Reproduce

1. Create a new .net maui project
2. Add a class MainPageViewModel and define a bindable property "EnableLabel"
3. Im Mainpage.xaml create a label inside a stacklayout(or any) and bind  it like this

`
<Label Text="im hidden" IsVisible="{OnIdiom Default=false,Phone={Binding EnableLabel}}"/>
`

4.Set the BindingContext in Mainpage.xaml.cs to MainPageViewModel
5.In MainPageViewModel constructor set EnableLabel to true

Build and deploy

Expected result : 
Wee se the label in phone in release mode

Actual result:
App crashes in release mode . Works fine in debug mode

### Link to public reproduction project repository

https://github.com/dotnet/maui/files/12414955/16725.zip

### Version with bug

7.0.49

### Is this a regression from previous behavior?

Yes, this used to work in Xamarin.Forms

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, Android, I was *not* able test on other platforms

### Affected platform versions

Android 13,Ios 16

### Did you find any workaround?

Try to assign the IsVisble property in xaml.cs file instead of binding like below
```
 if(DeviceInfo.Idiom == DeviceIdiom.Phone)
{
    myControl.IsVisible = true;
}
```

Note* - myControl is x:Name of your control

or if you don't want assign it in xaml.cs bind Isvisible property without Onidiom markup extension and in viewmodel where ever you are assigning the property value use 
```
 if(DeviceInfo.Idiom == DeviceIdiom.Phone)
{
    LabelVisible = true;
}
else
{
  LabelVisible = false;
}
```

**Comment by drasticactions** [comment]
Having a GitHub repo with this code makes it easier for us to quickly debug, verify, and potentially fix the issue against a codebase that you can also test against to verify our changes, as well as use it to create test cases against in the future. I can understand that it can be tedious to do, but it will be helpful.

**Comment by the-crow-club** [comment]
@drasticactions Sorry but i cant provide git hub repo due to my limited access to github at my workspace

**Comment by the-crow-club** [comment]
@XamlTest a bit of help here?

**Comment by XamlTest** [comment]
I can verify this issue on Visual Studio Enterprise 17.8.0 Preview 1.0. Repro on Android 13.0-API33 and iOS .NET 8 with below Project:
[16725.zip](https://github.com/dotnet/maui/files/12414955/16725.zip)

**Comment by hwsmo** [comment]
Hi, I can also confirm the bug. It cost me a lot of nerves until I found out that this was the case. Very frustrating :(

**Comment by formerlymisterhenson** [comment]
Facing the same issue - as a workaround I created a converter (which basically does the conditional part mentionend by op)

**Comment by StephaneDelcroix** [comment]
should be fixed by #24594. writing a test
